sdk/include/ndk/extypes.h

```diff
@@ -1532,6 +1554,17 @@ typedef struct _SYSTEM_MEMORY_LIST_INFORMATION
     SIZE_T ModifiedPageCountPageFile;
 } SYSTEM_MEMORY_LIST_INFORMATION, *PSYSTEM_MEMORY_LIST_INFORMATION;
 
+//
+// Firmware variable attributes
+//
+#define VARIABLE_ATTRIBUTE_NON_VOLATILE                             0x00000001
+#define VARIABLE_ATTRIBUTE_BOOTSERVICE_ACCESS                       0x00000002
+#define VARIABLE_ATTRIBUTE_RUNTIME_ACCESS                           0x00000004
+#define VARIABLE_ATTRIBUTE_HARDWARE_ERROR_RECORD                    0x00000008
+#define VARIABLE_ATTRIBUTE_AUTHENTICATED_WRITE_ACCESS               0x00000010
+#define VARIABLE_ATTRIBUTE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS    0x00000020
+#define VARIABLE_ATTRIBUTE_APPEND_WRITE                             0x00000040
```

Comment:
Those attributes are not exposed by the standard wdk in the MS DDK (or even in the PSDK)?
If they are exposed then they should be moved to the xdk template files.

Reply:
No, they introduced in MSDoc only (and in UEFI Specification named EFI_VARIABLE_XXX), WDK/DDK didn't expose them.